sql/hive-thriftserver/src/test/scala/org/apache/spark/sql/hive/thriftserver/HiveThriftServer2Suites.scala

```diff
@@ -811,6 +811,61 @@ class HiveThriftBinaryServerSuite extends HiveThriftJdbcTest {
       }
     }
   }
+
+  test("SPARK-31859 Thriftserver works with spark.sql.datetime.java8API.enabled=true") {
+    withJdbcStatement() { statement =>
+      withJdbcStatement() { st =>
```

Comment:
@juliuszsompolski Did you use this nested `st` intentionally? Looks like `statement` is not used.

Reply:
.... this was not intentional, looks like a copy paste error on my part. I raised https://github.com/apache/spark/pull/28735/ to fix it.